python/cudf/cudf/core/column/numerical.py

```diff
@@ -320,7 +330,20 @@ def find_and_replace(
         Return col with *to_replace* replaced with *value*.
         """
         to_replace_col = column.as_column(to_replace)
+        if to_replace_col.null_count == len(to_replace_col):
+            # If all of `to_replace` are `None`, dtype of `to_replace_col`
```

Comment:
Since this typecasting is happening for the same reason in both cases IMO we really only need one comment explaining why.

Reply:
Yup, moved to a single comment